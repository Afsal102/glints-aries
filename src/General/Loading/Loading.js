/* @flow */

import React from 'react';
import { LoadingContainer, Spinner } from '../../Style/General/LoadingStyle';

const Loading = (props: Props) => {
  const {
    size,
    className,
  } = props;

  return (
    <LoadingContainer className={className}>
      <Spinner size={size} />
    </LoadingContainer>
  );
};

type Props = {
  size: string,
  className: string,
};

export default Loading;
