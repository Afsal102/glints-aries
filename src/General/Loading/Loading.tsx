import * as React from 'react';
import { LoadingContainer, Spinner } from '../../Style/General/LoadingStyle';

const Loading: React.FunctionComponent<Props> = (props) => {
  const {
    className,
  } = props;

  return (
    <LoadingContainer
      id="aries-loading"
      className={className}
      role="alert"
      aria-busy="true"
      aria-live="polite"
    >
      <Spinner />
    </LoadingContainer>
  );
};

interface Props {
  className?: string;
}

export default Loading;
