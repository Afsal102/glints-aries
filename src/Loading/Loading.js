/* @flow */
import React from 'react';
import Label from '../Label';
import { LoadingContainer, Spinner } from './../Style/LoadingStyle';

const Loading = (props: Props) => {
  const { size, text } = props;

  return (
    <LoadingContainer>
        <Spinner size={size} />
        {text && 
            <Label size="medium" theme="red">{ text }</Label>
        }
        
    </LoadingContainer>
  );
}

export default Loading;
