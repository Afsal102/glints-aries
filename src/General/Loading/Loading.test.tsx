import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Loading from './Loading';

const loading = <Loading />;

describe('<Loading>', () => {
  it('should render as expected', () => {
    const loadingSnapshot = renderer.create(loading).toJSON();
    expect(loadingSnapshot).toMatchSnapshot();
  });
});
