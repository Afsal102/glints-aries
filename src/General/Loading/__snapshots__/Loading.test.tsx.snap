// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading> should render as expected 1`] = `
<div
  aria-busy="true"
  aria-live="polite"
  className="LoadingStyle__LoadingContainer-fg2qx3-0 ioHJOi aries-loading"
  data-testid="alert"
  role="alert"
>
  <div
    className="LoadingStyle__Spinner-fg2qx3-1 fQRrhB"
  />
</div>
`;
